src/views/Generator/TestRuleContainer/TestRuleContainer.tsx

```diff
@@ -25,25 +31,72 @@ export function TestRuleContainer() {
   }
 
   return (
-    <ScrollArea scrollbars="vertical">
-      <StickyPanelHeader>
-        <RulesNotAppliedCallout />
-
-        <Flex align="center" gap="3">
-          <Heading size="2" weight="medium">
-            Test rules ({rules.length})
-          </Heading>
-          <NewRuleMenu />
-        </Flex>
-      </StickyPanelHeader>
-      <SortableRuleList rules={rules} onSwapRules={swapRules} />
-      {shouldShowHelpMessage && (
-        <EmptyMessage
-          message="Configure your test logic by adding a new rule"
-          pb="2"
-          action={<NewRuleMenu variant="solid" size="2" color="orange" />}
+    <>
+      <ScrollArea scrollbars="vertical">
+        <StickyPanelHeader>
+          <RulesNotAppliedCallout />
+
+          <Flex align="center" gap="3">
+            <Heading size="2" weight="medium">
+              Test rules ({rules.length})
+            </Heading>
+            <NewRuleMenu />
+            <Feature feature="auto-correlation">
+              <Button
+                variant="ghost"
+                size="1"
+                color="gray"
+                onClick={() => setIsAutoCorrelationDialogOpen(true)}
+              >
+                <WandSparkles />
+                Auto correlate
+              </Button>
+            </Feature>
+          </Flex>
+        </StickyPanelHeader>
+        <SortableRuleList rules={rules} onSwapRules={swapRules} />
+        {shouldShowHelpMessage && (
+          <>
+            <FeatureDisabled feature="auto-correlation">
+              <EmptyMessage
+                message="Configure your test logic by adding a new rule"
+                pb="2"
+                action={<NewRuleMenu variant="solid" size="2" color="orange" />}
+              />
+            </FeatureDisabled>
+
+            <Feature feature="auto-correlation">
+              <EmptyMessage
+                message="Configure your test logic by adding a new rule"
+                pb="2"
+                action={
+                  <Flex gap="2" align="center">
+                    <Button
+                      onClick={() => setIsAutoCorrelationDialogOpen(true)}
+                    >
+                      <WandSparkles />
+                      Auto correlate
+                    </Button>{' '}
+                    <NewRuleMenu
+                      variant="solid"
+                      size="2"
+                      color="orange"
+                      text="Add rule manually"
```

Comment:
Changed button text as a way to promote auto correlation, "do it automatically OR do it manually"